server/src/main/java/org/elasticsearch/script/field/GeoPointDocValuesField.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.script.field;
+
+import org.apache.lucene.util.ArrayUtil;
+import org.elasticsearch.common.geo.GeoBoundingBox;
+import org.elasticsearch.common.geo.GeoPoint;
+import org.elasticsearch.index.fielddata.MultiGeoPointValues;
+import org.elasticsearch.index.fielddata.ScriptDocValues;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+public class GeoPointDocValuesField implements DocValuesField<GeoPoint>, ScriptDocValues.GeometrySupplier<GeoPoint> {
+
+    protected final MultiGeoPointValues input;
+    protected final String name;
+
+    protected GeoPoint[] values = new GeoPoint[0];
+    protected int count;
+
+    // maintain bwc by making centroid and bounding box available to ScriptDocValues.GeoPoints
+    private ScriptDocValues.GeoPoints geoPoints = null;
+    private final GeoPoint centroid = new GeoPoint();
+    private final GeoBoundingBox boundingBox = new GeoBoundingBox(new GeoPoint(), new GeoPoint());
+
+    public GeoPointDocValuesField(MultiGeoPointValues input, String name) {
+        this.input = input;
+        this.name = name;
+    }
+
+    @Override
+    public void setNextDocId(int docId) throws IOException {
+        if (input.advanceExact(docId)) {
+            resize(input.docValueCount());
+            if (count == 1) {
+                setSingleValue();
+            } else {
+                setMultiValue();
+            }
+        }
+    }
+
+    private void resize(int newSize) {
+        count = newSize;
+        if (newSize > values.length) {
+            int oldLength = values.length;
+            values = ArrayUtil.grow(values, count);
+            for (int i = oldLength; i < values.length; ++i) {
+                values[i] = new GeoPoint();
+            }
+        }
+    }
+
+    private void setSingleValue() throws IOException {
+        GeoPoint point = input.nextValue();
```

Comment:
Can you remove this from [the `Supplier`](https://github.com/elastic/elasticsearch/pull/81395/files#diff-a56cff73d443a82dd1e5368b0fce2ef82d0aaa9ba10ba332dcd8cac11037f872R321)?

Reply:
Removed this supplier as it's no longer necessary since there are no other dependencies on it.